pkg/controllers/provisioning/v1alpha1/reallocator/controller.go

```diff
@@ -53,31 +55,60 @@ func (c *Controller) Name() string {
 func NewController(kubeClient client.Client, cloudProvider cloudprovider.Factory) *Controller {
 	return &Controller{
 		filter:        &Filter{kubeClient: kubeClient},
+		collector:     &Collector{kubeClient: kubeClient},
 		cloudProvider: cloudProvider,
 	}
 }
 
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(object controllers.Object) error {
+	provisioner := object.(*v1alpha1.Provisioner)
 	ctx := context.TODO()
-	// 1. Filter all nodes with Karpenter TTL label
-	// TODO: Filter only nodes with the Provisioner label
 
-	// 2. Remove TTL from nodes with TTL label that have pods and if past TTL, cordon/drain node
+	// 1. Filter all nodes with Karpenter labels
+	ttlNodes, err := c.filter.getNodesWithLabels(ctx, []string{
```

Comment:
Btw, I tend to make my methods public if they're intended to be a public API, even if they happen to be in the same package. Imagine the filtering logic gets so complex that you need to move it to another package. At that point, you'd need to make all of it's methods public. It's also a nice signal to the reader of which methods should be used by other files vs which ones are internally file scoped.

Reply:
Got it. I'll do that. I like that pattern.